src/HttpParser.jl

```diff
@@ -87,6 +87,59 @@ type ParserSettings
     on_chunk_complete::Ptr{Void}
 end
 
+function ParserSettings(on_message_begin_cb::Ptr{Void},
+  on_url_cb::Ptr{Void},
+  on_status_complete_cb::Ptr{Void},
+  on_header_field_cb::Ptr{Void},
+  on_header_value_cb::Ptr{Void},
+  on_headers_complete_cb::Ptr{Void},
+  on_body_cb::Ptr{Void},
+  on_message_complete_cb::Ptr{Void})
+    Base.depwarn("ParserSettings(on_url_cb, on_status_complete_cb, "
+            * " on_header_field_cb, on_header_value_cb, on_header_value_cb,"
+            * " on_headers_complete_cb, on_body_cb, on_message_complete_cb)"
+            * " is deprecated use ParserSettings(on_url_cb, on_status_complete_cb, "
+            * " on_header_field_cb, on_header_value_cb, on_header_value_cb,"
+            * " on_headers_complete_cb, on_body_cb, on_message_complete_cb,"
+            * " on_chunk_header, on_chunk_complete)", :ParserSettings)
+    function dummy(parser)
+      0
+    end
+    dummy_cb = cfunction(dummy, HTTP_CB...)
+    ParserSettings(on_message_begin_cb, on_url_cb, on_status_complete_cb,
+      on_header_field_cb, on_header_value_cb, on_headers_complete_cb,
+      on_body_cb, on_message_complete_cb, dummy_cb, dummy_cb)
+end
+
+function ParserSettings(;
+  on_message_begin_cb::Ptr{Void} = C_NULL,
+  on_url_cb::Ptr{Void} = C_NULL,
+  on_status_complete_cb::Ptr{Void} = C_NULL,
+  on_header_field_cb::Ptr{Void} = C_NULL,
+  on_header_value_cb::Ptr{Void} = C_NULL,
+  on_headers_complete_cb::Ptr{Void} = C_NULL,
+  on_body_cb::Ptr{Void} = C_NULL,
+  on_message_complete_cb::Ptr{Void} = C_NULL,
+  on_chunk_header::Ptr{Void} = C_NULL,
+  on_chunk_complete::Ptr{Void} = C_NULL)
+    function dummy(x...)
+      0
+    end
+    dummy_cb = cfunction(dummy, HTTP_CB...)
+    dummy_data_cb = cfunction(dummy, HTTP_DATA_CB...)
```

Comment:
> If you look at this short example you will see that the generated c function pointer is always the same, which implies the opposite.

I mean the dummy in the two functions you add. Calling `cfunction` on the same function will reuse the c pointer and syntactically equivalent function definition shares the code but that's not what I mean.